### PR TITLE
Lesson-1-6

### PR DESCRIPTION
**Pull request by leipei92**
Week 1-6

Added remove button/Autofocus on each render

### CODE REVIEW

File: src/AddTodoForm.js [hunk 1]
```diff
@@ -9,34 +10,36 @@ const AddTodoForm = ({ onAddTodo }) => {
     setTodoTitle(newTodoTitle);
   }
 
-
   const handleAddTodo = (event) => {
     event.preventDefault();
     if (!todoTitle) {
-      throw new Error("Can't be empty") 
+      throw new Error("Can't be empty")
```
Comment (JoHeyJo) [inline src/AddTodoForm.js]: It's great that you're thinking about this! You will soon need to start gracefully handling all errors so it doesn't break your app. This still needs a bit of work but it's a good start.

File: src/App.js [hunk 2]
```diff
@@ -22,6 +22,14 @@ const useSemiPersistentState = () => {
 function App() {
 
   const [todoList, setTodoList] = useSemiPersistentState()
+
+  const removeTodo = (id) => {
```
Comment (JoHeyJo) [inline src/App.js:26]: Nicely done getting this filter function to work!

File: src/InputWithLabel.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+
```
Comment (JoHeyJo) [inline src/InputWithLabel.js:1]: You have the right idea here. 
This component is abstract but is everything connected and working properly? I would suggest going over the directions and implementing **exactly** what's described and then after everything is working you can consider further refactoring the component.

File: src/InputWithLabel.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    type,
+    name,
+    value,
+    isFocused,
+    onChange,
+}) => {
+
+    const inputRef = React.useRef();
+
+    React.useEffect(() => {
```
Comment (JoHeyJo) [inline src/InputWithLabel.js:15]: Nice job handling `useRef` this is working as intended.

File: src/InputWithLabel.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+
+import React from 'react';
+
+const InputWithLabel = ({
```
Comment (JoHeyJo) [inline src/InputWithLabel.js:4]: This sort of looks like what was shown in the book. Let's just focus on what the directions suggest.

File: src/TodoListItem.js [hunk 6]
```diff
@@ -1,10 +1,17 @@
...
 
-const TodoListItem = ({ todo}) => {
+const TodoListItem = ({ todo, onRemoveTodo }) => {
     return (
-        <li>{todo}</li>
+        <>
+            <li>
+                {todo.title}
+                <button type="button"
+                    onClick={() => onRemoveTodo(todo.id)}> Remove </button>
```
Comment (JoHeyJo) [inline src/TodoListItem.js:9]: This is great! Good use of a one-liner.

File: src/InputWithLabel.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+    React.useEffect(() => {
+
+        inputRef.current.focus();
+
+    });
+
+    return (
+        <>
+            <label htmlFor={id} > {children} </label>
```
Comment (JoHeyJo) [inline src/InputWithLabel.js:23]: Your label and input are not connected.

**Review by JoHeyJo — CHANGES_REQUESTED** [review]
Nice job getting your app to work after adding additional functionality and creating a reusable component! 

Even though your application is working there is a small bug that affects the user experience and your `InputWithLabel` needs some polishing up.

I encourage you to use GitHub for collabing on code reviews so comment directly on the code you need any help with.

File: src/InputWithLabel.js (hunk 7 above)
Comment (leipei92) [reply src/InputWithLabel.js:23]: If I add the same ID in my input will that help?

Comment (JoHeyJo) [reply src/InputWithLabel.js:23]: @leipei92 correct. But be sure that you're assigning that `id` to the correct attribute.

Comment (leipei92) [reply src/InputWithLabel.js:23]: I have made lots of changes. May I merge the pull request now? It seems to be working as should be now.

**Review by JoHeyJo — APPROVED** [review]
@leipei92 looks good to me, nicely done! 
Before merging clean up your code a bit. There is a lot of empty space in the middle of your functions code block and in some cases there is no space between the end of a function and beginning of a new one. This is essential for clean, legible code.

**Comment by JoHeyJo** [comment]
@leipei92 can you resubmit this assignment on the CTD page? I need to approve it. Thanks!

**Comment by leipei92** [comment]
All done!Thank you for the quick response.

**Pull request merged**